client/kafka/v2/kafka.go

```diff
@@ -0,0 +1,143 @@
+package v2
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/Shopify/sarama"
+	patronerrors "github.com/beatlabs/patron/errors"
+	"github.com/beatlabs/patron/internal/validation"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+const (
+	asyncProducerComponent = "kafka-async-producer"
+	syncProducerComponent  = "kafka-sync-producer"
+	messageCreationErrors  = "creation-errors"
+	messageSendErrors      = "send-errors"
+	messageSent            = "sent"
+)
+
+var messageStatus *prometheus.CounterVec
+
+func init() {
+	messageStatus = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Namespace: "client",
+			Subsystem: "kafka_producer",
+			Name:      "message_status",
+			Help:      "Message status counter (produced, encoded, encoding-errors) classified by topic",
+		}, []string{"status", "topic", "type"},
+	)
+
+	prometheus.MustRegister(messageStatus)
+}
+
```

Comment:
are status and deliveryType values limited to a specific number ?
Maybe we could use enums or predefined string properties ?

Reply:
Done

Reply:
i had to revert one of the values as it broke the tests. the reason is that they are internals and integration tests do not run inside the same package.